swarm.go

```diff
@@ -374,11 +374,21 @@ func (s *Swarm) Start(srv *p2p.Server) error {
 	log.Info("Updated bzz local addr", "oaddr", fmt.Sprintf("%x", newaddr.OAddr), "uaddr", fmt.Sprintf("%s", newaddr.UAddr))
 
 	if s.config.SwapEnabled {
-		err := s.swap.Deploy(context.Background(), s.backend, s.config.Path)
-		if err != nil {
-			return fmt.Errorf("Unable to deploy swap contract: %v", err)
+		if s.config.Contract != (common.Address{}) {
+			address := s.config.Contract
+			if err := cswap.ValidateCode(context.Background(), s.backend, address); err != nil {
```

Comment:
shouldn't we call `verifyContract` instead of `ValidateCode` directly?

Reply:
Nice one. Indeed that would be better!

Reply:
Currently, `verifyContract` is not exported. I solved it by exporting verifyContract.

Reply:
while i see your point, i think changes like these should be made in an atomic manner—i'm not a fan of leaving code around due to "_just in case_"- or "_for later use_"-type of situations.

right now, all `verifyContract` does is call `ValidateCode`. even their comments are very similar:
```
// verifyContract checks if the bytecode found at address matches the expected bytecode
```
```
// ValidateCode checks that the on-chain code at address matches the expected swap
// contract code.
```
i would either: 
1. remove `ValidateCode` and always call `verifyContract` directly
2. always call `ValidateCode` for consistency's sake

i will not consider this a blocker for approval, though :+1:

Reply:
Choose for option 2. 
Meant I also had to rewrite some tests:
`TestVerifyContract` => `TestValidateCode`
`TestVerifyWrongContract` => `TestValideWrongCode`
I don't find this particularly nice, as the unit tests in `swap` are now actually testing functionality of `contracts/swap`. If we would like to change this, however, there is a need to create a new testSuite in `contracts/swap`. 

I refrain from doing this at the moment but will put it on the agenda for the next sync meeting to hear to the opinion of team members.

Reply:
thank you for putting this down for discussion. 

i'm all for a new test suite if the situation calls for it :+1: